pycheribuild/__main__.py

```diff
@@ -168,7 +168,7 @@ def real_main():
             docker_dir_mappings = [
                 # map cheribuild and the sources read-only into the container
                 "-v", cheribuild_dir + ":/cheribuild:ro",
-                "-v", str(cheri_config.source_root.absolute()) + ":/source:ro",
+                "-v", str(cheri_config.source_root.absolute()) + ":/source",
```

Comment:
Ideally, the source tree could remain read-only. I guess this is only required for GDB?

Reply:
Ah yes sorry! This is needed because the script clones git repositories into the /source directory - if it's set to read-only, the operation fails, I forgot to put this in the commit message, I'll fix that up with the other stuff.

I suppose ideally the cloning would be done *before* starting up docker, but that's currently not the way it's done, I'll take a look and see how easy modifying that would be.

Reply:
Hm, definitely doesn't look trivially easy to do, as the docker wrapper is only really understood at the top level of the build system, while the cloning is done later during compilation of the various projects. It might be worth revisiting this another time, but I think it might be a bit complex for this set of fixes?

Reply:
That obviously makes sense.
I only ever used the docker wrapper with already-cloned repositories, so I never hit that issue and added the RO flag to ensure that the source directory remains the same. But without major refactoring, we will have to continue cloning inside so removing the flag is the best solution for now.